lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,435 @@
+            dataset.peek = "Flow Stats1 file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Flow Stats1 file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            first_header = f.readline().strip().split("\t")[0]
+            if first_header != "FileID":
+                return False
+            return True
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'text/tab-separated-values'
+
+
+class FlowStats2(Tabular):
+    """Class describing a Flow Stats file"""
+    file_ext = "flowstat2"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Flow Stats2 file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Flow Stats2 file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            smp_name = f.readline().strip().split("\t")[-1]
+            if smp_name != "SampleName":
+                return False
+            return True
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'text/tab-separated-values'
+
+
+class FlowStats3(Tabular):
+    """Class describing a Flow Stats file"""
+    file_ext = "flowstat3"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Flow Stats3 file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Flow Stats3 file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            last_col = f.readline().strip().split("\t")[-1]
+            if last_col != "Percentage_stdev":
+                return False
+            values = f.readline().strip().split("\t")
+            for vals in values:
+                if not is_number(vals):
+                    return False
+            return True
+
+    def get_mime(self):
```

Comment:
Is this a valid mime type? Why overwrite the Foo from tabular?

Reply:
Not sure why the person who wrote this did it, I agree it is unnecessary and I have removed it, thanks for spotting it!